go/vt/vttablet/tabletserver/repltracker/writer.go

```diff
@@ -117,14 +149,12 @@ func (w *heartbeatWriter) InitDBConfig(target *querypb.Target) {
 // Open sets up the heartbeatWriter's db connection and launches the ticker
 // responsible for periodically writing to the heartbeat table.
 func (w *heartbeatWriter) Open() {
-	if !w.enabled {
-		return
-	}
 	w.mu.Lock()
 	defer w.mu.Unlock()
 	if w.isOpen {
 		return
 	}
+	w.isOpen = true
```

Comment:
IMO we should keep this at the end of the function

Reply:
Now using `defer` to ensure the value is changed (there's early `return`s later on in the function)